File: java/src/main/java/com/databricks/zerobus/IPCCompressionType.java
package com.databricks.zerobus;

/**
 * Compression type for Arrow IPC messages.
 *
 * <p>When set on {@link ArrowStreamConfigurationOptions}, the SDK compresses each Arrow record
 * batch before transmitting it over the Arrow Flight connection. This can significantly reduce
 * network bandwidth at the cost of additional CPU usage.
 *
 * <p>Supported codecs:
 *
 * <ul>
 *   <li>{@link #NONE} - No compression (default)
 *   <li>{@link #LZ4_FRAME} - LZ4 frame compression (fast, moderate ratio)
 *   <li>{@link #ZSTD} - Zstandard compression (slower, better ratio)
 * </ul>
 *
 * @see ArrowStreamConfigurationOptions.ArrowStreamConfigurationOptionsBuilder#setIpcCompression
 */
public enum IPCCompressionType {
  /** No compression. This is the default. */
  NONE,

  /** LZ4 frame compression. Offers fast compression with a moderate compression ratio. */
  LZ4_FRAME,

  /** Zstandard compression. Offers slower compression with a better compression ratio. */
  ZSTD
}

File: dotnet/src/Databricks.Zerobus/IPCCompressionType.cs
namespace Databricks.Zerobus;

/// <summary>
/// Compression type for Apache Arrow Flight IPC messages.
/// </summary>
public enum IPCCompressionType
{
    /// <summary>No compression (default).</summary>
    None = -1,

    /// <summary>LZ4 Frame compression.</summary>
    Lz4Frame = 0,

    /// <summary>Zstandard compression.</summary>
    Zstd = 1
}
